Generated from spruceUI/PPSSPP-spruce@c1c560e:patches/common/fullscreen.py against hrydgard/ppsspp
d357e6a32934 (v1.20.3) with common patches 1-0 applied (2026-10-12). Apply in order.

diff --git a/SDL/SDLMain.cpp b/SDL/SDLMain.cpp
index 5eea6e5..704af37 100644
--- a/SDL/SDLMain.cpp
+++ b/SDL/SDLMain.cpp
@@ -827,9 +827,9 @@ void UpdateWindowState(SDL_Window *window) {
 
 		Uint32 window_flags = SDL_GetWindowFlags(window);
 		if (g_Config.bFullScreen) {
-			window_flags |= SDL_WINDOW_FULLSCREEN_DESKTOP;
+			window_flags |= SDL_WINDOW_FULLSCREEN;
 		} else {
-			window_flags &= ~SDL_WINDOW_FULLSCREEN_DESKTOP;
+			window_flags &= ~SDL_WINDOW_FULLSCREEN;
 		}
 		SDL_SetWindowFullscreen(window, window_flags);
 	}
@@ -959,7 +959,7 @@ static void ProcessSDLEvent(SDL_Window *window, const SDL_Event &event, InputSta
 			Native_NotifyWindowHidden(false);
 
 			Uint32 window_flags = SDL_GetWindowFlags(window);
-			bool fullscreen = (window_flags & (SDL_WINDOW_FULLSCREEN | SDL_WINDOW_FULLSCREEN_DESKTOP));
+			bool fullscreen = (window_flags & (SDL_WINDOW_FULLSCREEN | SDL_WINDOW_FULLSCREEN));
 
 			// This one calls NativeResized if the size changed.
 			Native_UpdateScreenScale(new_width_px, new_height_px, UIScaleFactorToMultiplier(g_Config.iUIScaleFactor));
@@ -989,7 +989,7 @@ static void ProcessSDLEvent(SDL_Window *window, const SDL_Event &event, InputSta
 		case SDL_WINDOWEVENT_MOVED:
 		{
 			Uint32 window_flags = SDL_GetWindowFlags(window);
-			bool fullscreen = (window_flags & (SDL_WINDOW_FULLSCREEN | SDL_WINDOW_FULLSCREEN_DESKTOP));
+			bool fullscreen = (window_flags & (SDL_WINDOW_FULLSCREEN | SDL_WINDOW_FULLSCREEN));
 			if (!fullscreen) {
 				g_Config.iWindowX = (int)event.window.data1;
 				g_Config.iWindowY = (int)event.window.data2;
@@ -1469,7 +1469,7 @@ int main(int argc, char *argv[]) {
 	Uint32 mode = 0;
 	for (int i = 1; i < argc; i++) {
 		if (!strcmp(argv[i], "--fullscreen")) {
-			mode |= SDL_WINDOW_FULLSCREEN_DESKTOP;
+			mode |= SDL_WINDOW_FULLSCREEN;
 			g_Config.DoNotSaveSetting(&g_Config.bFullScreen);
 		} else if (set_xres == -2)
 			set_xres = parseInt(argv[i]);
@@ -1562,19 +1562,19 @@ int main(int argc, char *argv[]) {
 
 	// Force fullscreen if the resolution is too low to run windowed.
 	if (g_DesktopWidth < 480 * 2 && g_DesktopHeight < 272 * 2) {
-		mode |= SDL_WINDOW_FULLSCREEN_DESKTOP;
+		mode |= SDL_WINDOW_FULLSCREEN;
 	}
 
 	// If we're on mobile, don't try for windowed either.
 #if defined(MOBILE_DEVICE) && !PPSSPP_PLATFORM(SWITCH)
 	mode |= SDL_WINDOW_FULLSCREEN;
 #elif defined(USING_FBDEV) || PPSSPP_PLATFORM(SWITCH)
-	mode |= SDL_WINDOW_FULLSCREEN_DESKTOP;
+	mode |= SDL_WINDOW_FULLSCREEN;
 #else
 	mode |= SDL_WINDOW_RESIZABLE | SDL_WINDOW_ALLOW_HIGHDPI;
 #endif
 
-	if (mode & SDL_WINDOW_FULLSCREEN_DESKTOP) {
+	if (mode & SDL_WINDOW_FULLSCREEN) {
 		g_display.pixel_xres = g_DesktopWidth;
 		g_display.pixel_yres = g_DesktopHeight;
 		g_Config.bFullScreen = true;
@@ -1637,7 +1637,7 @@ int main(int argc, char *argv[]) {
 
 	// Use the setting from the config when initing the window.
 	if (g_Config.bFullScreen) {
-		mode |= SDL_WINDOW_FULLSCREEN_DESKTOP;
+		mode |= SDL_WINDOW_FULLSCREEN;
 		g_display.pixel_xres = g_DesktopWidth;
 		g_display.pixel_yres = g_DesktopHeight;
 	}
